src/stats.jl

```diff
@@ -379,8 +258,8 @@ function summarystats(
     # Subset the chain.
     _chains = Chains(chains, _clean_sections(chains, sections))
 
-    # Caluclate ESS separately.
-    ess_df = ess(_chains; sections = nothing, maxlag = maxlag)
+    # Calculate ESS separately.
+    ess_df = ess(_chains; sections = nothing, maxlag = maxlag) # this is a ChainDataFrame - maybe sections = _default_sections() by default
```

Comment:
Probably we should also allow to choose the ESS method here, in case you want the summary but not the default ESS algorithm (because it's too slow or whatever). Maybe just add an additional keyword argument - or maybe it's easier to specify the keyword arguments for `mcse` explicitly? I don't remember which arguments one can pass to that function.

Reply:
I think it's simpler to add `method` as an explicit keyword argument of `summarystats` with default value `ESSMethod()`, and to call `ess` with `method = method`?

Reply:
Sure we can do that but I was wondering what we actually pass to `mcse` (and in particular if it's less than two keyword arguments) since if we add `method` as keyword argument we specify two keyword arguments (and their defaults) for `ess`. In general, I don't like if default keyword arguments are repeated in multiple places (since one has to keep track that they don't diverge). It seems `mcse` actually takes at most one keyword argument (https://github.com/TuringLang/MCMCChains.jl/blob/dev/src/mcse.jl) and hence the following could be an alternative:
- Use dedicated method structs for `mcse` such that we can dispatch on them instead of checking symbols
- Add `size` as a parameter of the method for `:bm`
- Don't specify `maxlag` and `essmethod` in this method explicitly and just forward the remaining keyword arguments to `ess` (since no keyword arguments are required for `mcse` anymore)

Reply:
So, something like the following in `mcse.jl`

`abstract type AbstractEtype end

struct BM <: AbstractEtype 
    size :: Integer
    function BM(_size::Integer)
        size = _size
    end
end
BM() = BM(100)

struct IMSE <: AbstractEtype end

struct IPSE <: AbstractEtype end

function mcse(etype::BM, x::Vector{<:Real})
    size = etype.size
    ...
end

function mcse(::IMSE, x::Vector{<:Real})
    ...
end

function mcse(::IPSE, x::Vector{<:Real})
    ...
end`

and declaring `summarystats` in `stats.jl` as follows

`function summarystats(
    chains::Chains;
    sections = _default_sections(chains),
    append_chains::Bool = true,
    etype::AbstractEtype = IMSE(),
    kwargs... 
)`

then calling `ess_df = ess(_chains; sections = nothing, kwargs)`?

Reply:
Yep, along these lines. But I realized that it's probably best to go with your version for now, one could change it then in a separate PR.